src/main.rs

```diff
@@ -151,29 +151,28 @@ impl Halloy {
                     command.map(Message::Dashboard),
                 )
             }
-            Err(error) => match &error {
-                config::Error::Parse(_) | config::Error::LoadSounds(_) => (
-                    Screen::Help(screen::Help::new(error)),
-                    Config::default(),
-                    Task::none(),
-                ),
-                _ => {
+            Err(error) => {
+                if config::has_yaml_config() {
```

Comment:
We really should add a new `config::Error` variant for `config::Error::ConfigMissing { has_yaml_config: bool }` where we check if `path.exists` and if not, we can populate the error with `config::has_yaml_config`.

Then under this branch, we can simply check if error is `Error::ConfigMissing { has_yaml_config: true }` for `Screen::Migration`.

The `ConfigMissing { has_yaml_config: false }` can then route to `Screen::Welcome`. We only want to route here if the config is missing & no yaml config exits. Any parsing errors should truly route to `Screen::Help` w/ the error